x-pack/plugins/lens/public/editor_frame_service/editor_frame/frame_layout.tsx

```diff
@@ -18,67 +18,87 @@ export interface FrameLayoutProps {
-              {i18n.translate('xpack.lens.section.workspaceLabel', {
-                defaultMessage: 'Visualization workspace',
+          <EuiPageBody
+            restrictWidth={false}
+            className="lnsFrameLayout__pageContent"
+            aria-labelledby="lns_ChartTitle"
+          >
+            <section
+              className={classNames('lnsFrameLayout__sidebar lnsFrameLayout__sidebar--left', {})}
+              aria-labelledby="dataPanelId"
+            >
+              <EuiScreenReaderOnly>
+                <h2 id="dataPanelId">
+                  {i18n.translate('xpack.lens.section.dataPanelLabel', {
+                    defaultMessage: 'Data panel',
+                  })}
+                </h2>
+              </EuiScreenReaderOnly>
+              {props.dataPanel}
+            </section>
+            <section
+              className={classNames('lnsFrameLayout__pageBody', {
+                'lnsFrameLayout__pageBody-isFullscreen': isFullscreen,
               })}
-            </h2>
-          </EuiScreenReaderOnly>
-          {props.workspacePanel}
-          <div className="lnsFrameLayout__suggestionPanel">{props.suggestionsPanel}</div>
-        </section>
-        <section
-          className={classNames('lnsFrameLayout__sidebar lnsFrameLayout__sidebar--right', {
-            'lnsFrameLayout__sidebar-isFullscreen': isFullscreen,
-          })}
-          aria-labelledby="configPanel"
-        >
-          <EuiScreenReaderOnly>
-            <h2 id="configPanel">
-              {i18n.translate('xpack.lens.section.configPanelLabel', {
-                defaultMessage: 'Config panel',
+              aria-labelledby="workspaceId"
+            >
+              <EuiScreenReaderOnly>
+                <h2 id="workspaceId">
+                  {i18n.translate('xpack.lens.section.workspaceLabel', {
+                    defaultMessage: 'Visualization workspace',
+                  })}
+                </h2>
+              </EuiScreenReaderOnly>
+              {props.workspacePanel}
+              <div className="lnsFrameLayout__suggestionPanel">{props.suggestionsPanel}</div>
+            </section>
+            <section
+              className={classNames('lnsFrameLayout__sidebar lnsFrameLayout__sidebar--right', {
+                'lnsFrameLayout__sidebar-isFullscreen': isFullscreen,
               })}
-            </h2>
-          </EuiScreenReaderOnly>
-          {props.configPanel}
-        </section>
-      </EuiPageBody>
-    </EuiPage>
+              aria-labelledby="configPanel"
+            >
+              <EuiScreenReaderOnly>
+                <h2 id="configPanel">
+                  {i18n.translate('xpack.lens.section.configPanelLabel', {
+                    defaultMessage: 'Config panel',
+                  })}
+                </h2>
+              </EuiScreenReaderOnly>
+              {props.configPanel}
+            </section>
+          </EuiPageBody>
+        </EuiPage>
+      </EuiFlexItem>
+    </EuiFlexGroup>
```

Comment:
Rather than housing the container `section` for the warning callout in an `EuiFlexGroup` structure outside of `EuiPageBody`, would it make more sense to restructure things so that callout container `section` is nested within `EuiPageBody` (along with any necessary layout components)?

Better yet, would it be more ideal to display the field list sampling deprecation callout at the top of the field list itself, rather than generically at the top of the entire Lens application? Or are we hoping to build a single container for the possibility of displaying multiple deprecation messages? If the desire is indeed for a single container to house multiple message, would the Lens warning popover that conditionally appears above the visualization be a better and more scalable choice?

CCing @ghudgins and @flash1293 here, in case they have any thoughts.

Reply:
The way it's currently displayed is what we used in other places too to indicate breaking changes based on the users configuration in other apps (e.g. legacy renderers in agg based visualizations). I think it makes sense to keep it like this so we can use the container for other deprecations later on.

Reply:
yes, thanks Joe. My idea was precisely that in the future we will have the opportunity to reuse this mechanism.